core/src/Kokkos_Macros.hpp

```diff
@@ -382,6 +382,10 @@
 #define KOKKOS_IMPL_DEVICE_FUNCTION
 #endif
 
+#if !defined(KOKKOS_PRINTF)
+#define KOKKOS_PRINTF(format, ...) printf((format), __VA_ARGS__)
```

Comment:
The issue with variadic macros is _always_ going to be MSVC. I can never remember off of the top of my head what the differences are, but keep MSVC in mind before you go too far with it.

Reply:
I think the (main) issue is described in https://stackoverflow.com/questions/5134523/msvc-doesnt-expand-va-args-correctly. It seems that there are workarounds to be able to use a variadic macro from another variadic macro and also using this macro is only necessary if also the SYCL backend should be supported.